### PR TITLE
feat: add command to diff changed files for terraform

### PR DESCRIPTION
**Pull request by dcreey**
Tested:

Git Diff:
```
> git diff 93f94a90dc0e27a5e0df0bb3e4352d3a7b675532..a0d258869fcb98f25c15ebecff473fffa378a4c1 --name-only
terraform/gcp-org/x-projects/hyperchicken/guardian.deployments.tf
terraform/gcp-org/x-projects/nebula/guardian.deployments.tf
terraform/gcp-org/x-projects/tapestry/guardian.deployments.tf
terraform/modules/project/main.tf
```

Guardian Diff:
```
> guardian diff -base-ref 93f94a90dc0e27a5e0df0bb3e4352d3a7b675532 -head-ref a0d258869fcb98f25c15ebecff473fffa378a4c1 terraform
/usr/local/google/home/dcreey/dev/github/x-refinery-org/terraform/gcp-org/x-projects/hyperchicken
/usr/local/google/home/dcreey/dev/github/x-refinery-org/terraform/gcp-org/x-projects/hyperchicken/deployments/llm
/usr/local/google/home/dcreey/dev/github/x-refinery-org/terraform/gcp-org/x-projects/hyperchicken/deployments/pilot
/usr/local/google/home/dcreey/dev/github/x-refinery-org/terraform/gcp-org/x-projects/hyperchicken/workload-identity/github/theteamatx/x-hyperchicken-llm-infra
/usr/local/google/home/dcreey/dev/github/x-refinery-org/terraform/gcp-org/x-projects/hyperchicken/workload-identity/github/theteamatx/x-hyperchicken-pilot-infra
/usr/local/google/home/dcreey/dev/github/x-refinery-org/terraform/gcp-org/x-projects/nebula
/usr/local/google/home/dcreey/dev/github/x-refinery-org/terraform/gcp-org/x-projects/nebula/deployments/tidal-hw-exp
/usr/local/google/home/dcreey/dev/github/x-refinery-org/terraform/gcp-org/x-projects/nebula/experimental
/usr/local/google/home/dcreey/dev/github/x-refinery-org/terraform/gcp-org/x-projects/nebula/workload-identity/github/theteamatx/my-infra-repo
/usr/local/google/home/dcreey/dev/github/x-refinery-org/terraform/gcp-org/x-projects/nebula/workload-identity/github/theteamatx/x-nebula-ci
/usr/local/google/home/dcreey/dev/github/x-refinery-org/terraform/gcp-org/x-projects/nebula/workload-identity/github/theteamatx/x-nebula-tidal-hw-exp
/usr/local/google/home/dcreey/dev/github/x-refinery-org/terraform/gcp-org/x-projects/tapestry
/usr/local/google/home/dcreey/dev/github/x-refinery-org/terraform/gcp-org/x-projects/tapestry/deployments/main
```

### CODE REVIEW

File: pkg/commands/diff/command.go
Comment (drevell) [inline pkg/commands/diff/command.go:37]: I think this comment might have been copy-pasted without being updated.

File: pkg/terraform/terraform.go
Comment (drevell) [inline pkg/terraform/terraform.go:116]: Exported types should be commented

File: pkg/terraform/terraform.go
Comment (drevell) [inline pkg/terraform/terraform.go:117]: Might be very helpful to add comments here explaining what the inner and outer keys are in these maps (module names?)

File: pkg/terraform/terraform.go
Comment (drevell) [inline pkg/terraform/terraform.go:122]: Can strings be single-quoted in HCL? I seem to recall that they can, but I could be mistaken.

File: pkg/terraform/terraform.go
Comment (drevell) [inline pkg/terraform/terraform.go:205]: This can be simplfiied a bit with os.Readfile(), which saves opening and closing a file handle.

File: pkg/terraform/terraform.go
Comment (drevell) [inline pkg/terraform/terraform.go:230]: Might be worth inverting this match to save a level of indentation.

```
if len(ms) != 2 {
  continue
}
relativeModulePath := ...
```

File: pkg/terraform/terraform.go
Comment (drevell) [inline pkg/terraform/terraform.go:236]: I think we try to avoid the unnecessary `else`s like this that follow a `return` or `continue`.

**Review by drevell — CHANGES_REQUESTED** [review]

File: pkg/terraform/terraform.go
Comment (verbanicm) [inline pkg/terraform/terraform.go:183]: Can we use HCL parser instead of regex? Similar to https://github.com/abcxyz/guardian/blob/main/pkg/terraform/terraform.go#L123.

File: pkg/commands/diff/command.go
Comment (verbanicm) [inline pkg/commands/diff/command.go:1]: We already do a diff in the [initialize command](https://github.com/abcxyz/guardian/blob/main/pkg/commands/initialize/init.go), I am not sure we need both.

The point of initialize is to generate the directories that required plan/apply. It seems like this logic should be amended to that command rather than a new, separate command.

File: pkg/terraform/terraform.go
Comment (dcreey) [reply pkg/terraform/terraform.go]: Done

File: pkg/terraform/terraform.go
Comment (dcreey) [reply pkg/terraform/terraform.go]: Done

File: pkg/terraform/terraform.go
Comment (dcreey) [reply pkg/terraform/terraform.go]: Done

File: pkg/terraform/terraform.go
Comment (dcreey) [reply pkg/terraform/terraform.go]: Done

File: pkg/terraform/terraform.go
Comment (dcreey) [reply pkg/terraform/terraform.go]: Resolved by switching to hcl parser.

File: pkg/commands/diff/command.go
Comment (dcreey) [reply pkg/commands/diff/command.go]: Updated.

File: pkg/terraform/terraform.go
Comment (drevell) [inline pkg/terraform/terraform.go:113]: This commented block looks like maybe you intended to remove it?

**Review by drevell — APPROVED** [review]

**Comment by drevell** [comment]
LGTM but I think we should get Mike's approval before merging.

File: pkg/terraform/terraform.go
Comment (dcreey) [reply pkg/terraform/terraform.go]: Done

**Review by drevell — DISMISSED** [review]

File: pkg/terraform/terraform_test.go
Comment (sethvargo) [inline pkg/terraform/terraform_test.go:298]:
```suggestion
func TestModules(t *testing.T) {
```

File: pkg/commands/diff/command.go
Comment (dcreey) [reply pkg/commands/diff/command.go:1]: Moved to initialize.

**Review by verbanicm — APPROVED** [review]

**Review by drevell — APPROVED** [review]

**Pull request merged**
